src/lock/file.rs

```diff
@@ -49,6 +50,8 @@ pub struct LockedExternalPlugin {
     pub files: Vec<PathBuf>,
     /// What templates to apply to each file.
     pub apply: Vec<String>,
+    /// Hooks executed during template evaluation.
+    pub hooks: Option<BTreeMap<String, String>>,
```

Comment:
Is there a good reason for this to be an `Option`? Might make the code a bit easier if you default to an empty `BtreeMap` in the `None` case.

Reply:
No good reason. I've changed to `BTreeMap<String, String>`.